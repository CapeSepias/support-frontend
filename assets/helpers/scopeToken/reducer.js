// @flow

// ----- Types ----- //

export type ScopeToken = ?string;


// ----- Setup ----- //

const initialState: ScopeToken =
  window.guardian && window.guardian.scopeToken ? window.guardian.scopeToken : null;


// ----- Reducer ----- //

export default function scopeTokenReducer(state: ScopeToken = initialState): ScopeToken {
  return state;
}
