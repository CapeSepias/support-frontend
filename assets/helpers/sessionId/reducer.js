// @flow
import uuidv4 from 'uuid';
import * as storage from 'helpers/storage';

// ----- Types ----- //

export type SessionId = ?string;


// ----- Setup ----- //

function getAndStoreId(keyName) {
  let value = storage.getSession(keyName);
  if (value === null) {
    value = uuidv4();
    storage.setSession(keyName, value);
  }
  return value;
}

const initialState: SessionId = getAndStoreId('sessionId');

// ----- Reducer ----- //

export default function sessionIdReducer(state: SessionId = initialState): SessionId {
  return state;
}
