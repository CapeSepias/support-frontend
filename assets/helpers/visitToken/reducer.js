// @flow

// ----- Imports ----- //

import type { Action } from './actions';


// ----- Types ----- //

export type VisitToken = {
    token: ?string,
};


// ----- Setup ----- //

const initialState: VisitToken = window.guardian && window.guardian.visitToken ? {
  token: window.guardian.visitToken,
} : {
  token: null,
};


// ----- Reducer ----- //

export default function visitTokenReducer(
  state: VisitToken = initialState,
  action: Action,
): VisitToken {

  switch (action.type) {

    case 'SET_VISIT_TOKEN':
      return Object.assign({}, state, { token: action.name });

    default:
      return state;

  }

}
