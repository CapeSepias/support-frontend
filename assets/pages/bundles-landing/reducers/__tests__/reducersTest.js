// @flow
import reducer from '../reducers';
import type { Contrib, PaperBundle, ContribState, Amount } from '../reducers';

describe('reducer tests', () => {

  const initialContrib: ContribState = {
    type: 'RECURRING',
    error: null,
    amount: {
      recurring: {
        value: '5',
        userDefined: false,
      },
      oneOff: {
        value: '25',
        userDefined: false,
      },
    },
  };
  const initialPaperBundle: PaperBundle = 'PAPER+DIGITAL';
  const initialState = { paperBundle: initialPaperBundle, contribution: initialContrib };

  it('should return the initial state', () => {

    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CHANGE_PAPER_BUNDLE', () => {

    const paperBundle: PaperBundle = 'PAPER';
    const action = {
      type: 'CHANGE_PAPER_BUNDLE',
      bundle: paperBundle,
    };

    expect(reducer(initialState, action).paperBundle).toEqual(paperBundle);
  });

  it('should handle CHANGE_CONTRIB_TYPE', () => {

    const contribType: Contrib = 'ONE_OFF';
    const action = {
      type: 'CHANGE_CONTRIB_TYPE',
      contribType,
    };

    expect(reducer(initialState, action).contribution.type).toEqual(contribType);
  });

  it('should handle CHANGE_CONTRIB_AMOUNT', () => {

    const amount: Amount = {
      value: '45',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT',
      amount,
    };

    expect(reducer(initialState, action).contribution.amount.recurring).toEqual(amount);
    expect(reducer(initialState, action).contribution.amount.oneOff).toEqual(amount);
  });

  it('should handle CHANGE_CONTRIB_AMOUNT_RECURRING', () => {

    const amount: Amount = {
      value: '45',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT_RECURRING',
      amount,
    };

    expect(reducer(initialState, action).contribution.amount.recurring).toEqual(amount);
    expect(reducer(initialState, action).contribution.amount.oneOff.value).toEqual('25');
  });

  it('should handle CHANGE_CONTRIB_AMOUNT_ONEOFF', () => {

    const amount: Amount = {
      value: '45',
      userDefined: true,
    };
    const action = {
      type: 'CHANGE_CONTRIB_AMOUNT_ONEOFF',
      amount,
    };

    expect(reducer(initialState, action).contribution.amount.oneOff).toEqual(amount);
    expect(reducer(initialState, action).contribution.amount.recurring.value).toEqual('5');
  });
});
