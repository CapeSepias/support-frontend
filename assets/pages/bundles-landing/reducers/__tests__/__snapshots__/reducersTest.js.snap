// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer tests should return the initial state 1`] = `
Object {
  "contribution": Object {
    "amount": Object {
      "oneOff": Object {
        "userDefined": false,
        "value": "25",
      },
      "recurring": Object {
        "userDefined": false,
        "value": "5",
      },
    },
    "error": null,
    "type": "RECURRING",
  },
  "intCmp": "",
}
`;
